azurerm/internal/services/apimanagement/api_management_resource.go

```diff
@@ -902,7 +902,7 @@ func resourceApiManagementServiceDelete(d *pluginsdk.ResourceData, meta interfac
 	}
 
 	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		if !response.WasNotFound(future.Response()) {
+		if future.FutureAPI == nil || !response.WasNotFound(future.Response()) {
 			return fmt.Errorf("waiting for deletion of API Management Service %q (Resource Group %q): %+v", name, resourceGroup, err)
 		}
```

Comment:
FWIW all of these "NotFound" checks in the Delete calls are something we'll be removing in the not too distant future, since as you've mentioned these are hiding configuration errors in Terraform Configs which need to be fixed

Reply:
>we'll be removing in the not too distant future,

Great!

>  these are hiding configuration errors in Terraform Configs which need to be fixed

Yes, but I think terraform should not crash and should give some error info.

Reply:
Terraform won't crash when these are removed fwiw, but that's outside of the scope of this PR :)